### PR TITLE
vim-patch:partial 2d8ed0203aed

### PR DESCRIPTION
**Pull request by dundargoc**
Update runtime files.
https://github.com/vim/vim/commit/2d8ed0203aedd5f6c22efa99394a3677c17c7a7a

Skip:
runtime/syntax/help.vim
runtime/syntax/vim.vim
runtime/doc/map.txt
runtime/doc/usr_51.txt
runtime/doc/usr_52.txt

### CODE REVIEW

**Comment by dundargoc** [comment]
There are too many previous patches missing to make anything useful out of this.

**Comment by clason** [comment]
you can mark as `vim-patch:partial <hash>` and comment on which patches are needed to port the skipped files. (Translations and desktop files are N/A; just ignore those; manual rewrite can be deferred as well.)

File: runtime/doc/syntax.txt [hunk 1]
```diff
@@ -4821,18 +4816,62 @@ in their own color.
...
+For further customization, such as changing |:highlight-link| associations,
+use another name, e.g.  "~/.config/nvim/colors/mine.vim", and use `:runtime`
+to load the original color scheme: >
+	runtime colors/evening.vim
+	hi Statement ctermfg=Blue guifg=Blue
+
+Before the color scheme will be loaded all default color list scripts
+(`colors/lists/default.vim`) will be executed and then the |ColorSchemePre|
+autocommand event is triggered.  After the color scheme has been loaded the
+|ColorScheme| autocommand event is triggered.
```
Comment (zeertzjq) [inline runtime/doc/syntax.txt:4836]: needs patch 8.2.3562

File: runtime/doc/syntax.txt [hunk 2]
```diff
@@ -4821,18 +4816,62 @@ in their own color.
...
+
+==============================================================================
+14. Highlight command			*:highlight* *:hi* *E28* *E411* *E415*
+
+There are three types of highlight groups:
+- The ones used for specific languages.  For these the name starts with the
+  name of the language.  Many of these don't have any attributes, but are
+  linked to a group of the second type.
+- The ones used for all syntax languages.
+- The ones used for the 'highlight' option.
```
Comment (zeertzjq) [inline runtime/doc/syntax.txt:4869]:
```suggestion
There are two types of highlight groups:
- The ones used for specific languages.  For these the name starts with the
  name of the language.  Many of these don't have any attributes, but are
  linked to a group of the second type.
- The ones used for all syntax languages.
```

File: runtime/colors/README.txt [hunk 3]
```diff
@@ -59,20 +63,20 @@ autocommand event.  For example:
...
-To see which highlight group is used where, see ":help highlight-groups" and
-":help group-name".
+To see which highlight group is used where, see `:help highlight-groups` and
+`:help group-name`.
 
 You can use ":highlight" to find out the current colors.  Exception: the
 ctermfg and ctermbg values are numbers, which are only valid for the current
 terminal.  Use the color names instead for better portability.  See
-":help cterm-colors".
+`:help cterm-colors`.
```
Comment (seandewar) [inline runtime/colors/README.txt:79]:
```suggestion
`:help cterm-colors` .
```

File: runtime/colors/README.txt [hunk 4]
```diff
@@ -59,20 +63,20 @@ autocommand event.  For example:
...
 
 	" load the "evening" colorscheme
 	runtime colors/evening.vim
 	" change the color of statements
 	hi Statement ctermfg=Blue guifg=Blue
 
-To see which highlight group is used where, see ":help highlight-groups" and
-":help group-name".
+To see which highlight group is used where, see `:help highlight-groups` and
+`:help group-name`.
```
Comment (seandewar) [inline runtime/colors/README.txt:74]: For proper highlighting, the space after the closing backtick is required (as the thing inside the backticks already has a space; yes it's weird)

```suggestion
`:help group-name` .
```

**Review by seandewar — COMMENTED** [review]
I'd also include https://github.com/vim/vim/commit/2d8ed0203aedd5f6c22efa99394a3677c17c7a7a#diff-e891ea60176f3f0bf0aa9bcd13c22cdb77f377ea932fe9515f5e0b16df8740daR1515 from `syntax.txt`.

Porting the color schemes section of `syntax.txt` is prolly gonna be a bit messy with all the `v:colornames` stuff (and a bit barren without it), so it's maybe best to skip for now indeed.

**Review by seandewar — APPROVED** [review]

**Pull request merged**